rts/Game/TraceRay.cpp

```diff
@@ -125,29 +128,34 @@ inline static bool TestTrajectoryConeHelper(
 	// (if object-distance is 0, tstPos == hitPos)
 	const float3 hitVec = tstDir * cvRelDst;
 	const float3 hitPos = (tstPos + hitVec) + (UpVector * (quadratic * cvRelDst * cvRelDst + linear * cvRelDst));
-
+	const float3 endPos = (tstPos + tstDir * length) + (UpVector * (quadratic * length * length + linear * length)); 
 	bool ret = false;
 
-	if (obj->GetBlockingMapID() < unitHandler.MaxUnits()) {
-		// first test the muzzle-position, then the impact-position
-		// (if neither is inside obstacle's CV, the weapon can fire)
-		ret = ret || ((cv->GetPointSurfaceDistance(static_cast<const CUnit*>(obj), nullptr, tstPos) - coneSize) <= 0.0f);
-		ret = ret || ((cv->GetPointSurfaceDistance(static_cast<const CUnit*>(obj), nullptr, hitPos) - coneSize) <= 0.0f);
-	} else {
-		ret = ret || ((cv->GetPointSurfaceDistance(static_cast<const CFeature*>(obj), nullptr, tstPos) - coneSize) <= 0.0f);
-		ret = ret || ((cv->GetPointSurfaceDistance(static_cast<const CFeature*>(obj), nullptr, hitPos) - coneSize) <= 0.0f);
+	CollisionQuery cq;
```

Comment:
Would be good to measure the perf implications of running full collision queries per shot.

Reply:
Compiled my side WITH this PR
![image](https://github.com/beyond-all-reason/spring/assets/44480662/a61e5361-ce2a-4c2c-9ff4-e2eddf5b7949)

Compiled my side WITHOUT this PR
![image](https://github.com/beyond-all-reason/spring/assets/44480662/4c5ea8c5-25fb-4bbd-b861-4bdebf58dc0d)

Not sure the proper way to benchmark (and the engine compiled on my side seems very slow), but this suggests no performance impact, and suggests a slight improvement.

Reply:
Also, if the weapon muzzle is inside a friendly unit, the shot is correctly blocked and does not fire.

Reply:
I see significant difference in `Sim::Unit::Weapon` occupation, does PR changes lie in the scope of that timer?